easy-coding-standard.yaml

```diff
@@ -39,16 +39,47 @@ services:
     # Some functions should not appear in the code
     PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff:
         forbiddenFunctions:
-            var_dump: null
-            dump: null
-            echo: null
-            require: null
-            require_once: null
+            # Program execution functions, inspired by https://github.com/spaze/phpstan-disallowed-calls
+            exec: null
+            passthru: null
+            proc_open: null
+            shell_exec: null
+            system: null
+            pcntl_exec: null
+            popen: null
+
+            # Dangerous function calls, inspired by https://github.com/spaze/phpstan-disallowed-calls
+            apache_setenv: null # might overwrite existing variables
+            dl: null # removed from most SAPIs, might load untrusted code
+            eval: null # eval is evil, please write more code and do not use eval()
+            extract: null # do not use extract() and especially not on untrusted data
+            highlight_file: null # might reveal source code or config files
+            pfsockopen: null # use fsockopen() to create non-persistent socket connections
+            posix_getpwuid: null # might reveal system user information
+            posix_kill: null # do not send signals to processes from the script
+            posix_mkfifo: null # do not create named pipes in the script
+            posix_mknod: null # do not create special files in the script
+            proc_nice: null # changes the priority of the current process
+            putenv: null # might overwrite existing variables
```

Comment:
Tady stimhle bych byl opatrny, dost se to pouziva

Reply:
Envy bys v aplikaci nemel vytvaret, jen prijimat.

Reply:
Máš nějaký příklad?

V pár specifických případech se to asi používá oprávněně (třeba v command-line tools), tam bych si to pak dal do výjimek.

Reply:
Jop vetsinou se jedna o command-line, nebo o testy.

treba priklad
```
$input = new ArgvInput();
if (null !== $_ENV['APP_ENV'] = $input->getParameterOption(['--env', '-e'], null, true)) {
    putenv('APP_ENV='.$_ENV['APP_ENV']);
    // force loading .env files when --env is defined
    $_SERVER['APP_ENV'] = null;
}
```

Reply:
Tohle ale bude v nějakém bootstrap.php nebo index.php, kde už je beztak i `require_once` (načítá se composer autoloader), takže to stejně bude ve výjimkách, ne?

Reply:
Nejenom tam, vyuziva to napr. SF komponenta Dotenv

Reply:
ok

Reply:
@hokypierce To jo, ale ty sám to ze svého kódu mimo nějaký index.php nevoláš, nebo jo? Kdyžtak mi hoď do Slacku link do bitbucket jestli to někde máte, kouknu.

Reply:
@OndraM mas pravdu ze to je jen z bootstrapu pri init app